src/main.cpp

```diff
@@ -382,16 +468,26 @@ void loop() {
 
   wdt_feed();
 
-#ifdef ESP32
+// TODO iotWebConf.doLoop() enabled - is there a reason why it was disabled?
+// #ifdef ESP32
   iotWebConf.doLoop();
-#endif
+// #endif
 
   if (WiFi.status() != WL_CONNECTED) {
-    reset();
+    lastConnectTime = 0;
+    // TODO call of reset() commented out in order to prevent restarts in listening mode
+    //reset();
+  }
+  else {
+    if (lastConnectTime == 0) {
+      lastConnectTime = millis();
+      reconnectCount++;
+    }
   }
 
   if (millis() > last_comms + 200*1000 ) {
-    reset();
+    // TODO call of reset() commented out in order to prevent restarts in listening mode
```

Comment:
maybe condition by running scheduler? or reset the value from scheduler?

Reply:
This was a real game changer in terms of restarts in listening mode. A connected message client never sent data and last_comms was therefore never updated. There was therefore a reboot every 20 minutes.

Reply:
I have tested the old code and it works. Maybe something went wrong during programming and testing (updating last_comms). I will undo the changes, but without any communication the adapter resets itself every 200 seconds for nothing. I don't think this is necessary behavior.